Refactor filter implementation and add new features

tanish-khurana opened the pull request:
-This implements a rate limiter for any probability exceeding 1 
-Also exportes data to category blocked stats

tanish-khurana committed:
Refactor filter implementation and add new features

-This implements a rate limiter for any probablity exceeding 1
-Also exposrts data to category blocked stats

tanish-khurana committed:
Update graph title and category options in dashboard

-This update dynamically updates the categories dropdown for the graph. No need to hard-code.

tanish-khurana committed:
Add category block graph feature to dashboard

mouper committed:
Merge branch 'main' into tanish-khurana-patch-1

mouper reviewed (changes requested):
Multiple errors, it is trying to import functions that don't exist.
<img width="1749" height="331" alt="image" src="https://github.com/user-attachments/assets/77c0a9db-9bd8-4449-919f-0522d5bfd424" />

tanish-khurana committed:
Remove batch_update_recent method from simple_filter

Removed the batch_update_recent method and its logic for filtering categories based on user settings.

tanish-khurana commented:
Changed the part of the code causing an error, pls have a look and let me know.

It-is-what-it-is1 commented:
Looks like a function is missing in the simple_filter.py.
<img width="1075" height="198" alt="Screenshot 2025-10-06 at 4 09 26 pm" src="https://github.com/user-attachments/assets/e0831d79-3e3d-4fdc-919a-c313fd6fc0cf" />

tanish-khurana committed:
Replace SimpleProbabilisticFilter with ProbabilisticDecayFilter

It was an API problem, filter.py file had a different name for the class. Should work now

tanish-khurana commented:
Pls check now, issue is resolved

It-is-what-it-is1 commented:
There is no error anymore but it appears that it is not updating the database. Is this just on my side?

keanscott commented:
> There is no error anymore but it appears that it is not updating the database. Is this just on my side?

it doesnt work on my end. doesnt seem to be any code to actually update the category_block_stats db. i will put some time in wednesday morning to try fix - unfortunately dont have time until then :(

tanish-khurana committed:
Replace SimpleProbabilisticFilter class with ProbabilisticDecayFilter

tanish-khurana commented:
<img width="948" height="573" alt="Screenshot 2025-10-07 at 1 18 10 AM" src="https://github.com/user-attachments/assets/058491d8-0592-4ff7-b4fb-82e3d6f2e5fb" />
<img width="948" height="255" alt="Screenshot 2025-10-07 at 1 18 00 AM" src="https://github.com/user-attachments/assets/534c4c07-41be-4f0f-9f2a-fafd7eb71378" />
I have tested it on my device, and both the databases are working, as well as being updated. It is also blocking the videos on a random basis, as we discussed. I would recommend going to the simplefilter file and bumping up the category strengths while testing. I did that, and the probability was actually increasing considerably.

keanscott committed:
fix: refactored file and db names, as well as changed watched_db and category_block_stats insertion logic.
- watched_db now inserts a new row when the same video is watched another time.
- category_block_stats inserts a new row when the level is updated.

keanscott committed:
chore: category_block_stats now decays each category present in the database when watched_db is updated.

keanscott committed:
fix: reverted changes to dashboard.

keanscott committed:
fix: reverted changes to dashboard.

keanscott reviewed (dismissed):
Everything now works. 
Changes to branch:
- reverted dashboard changes
- watched db now inserts a new row when an already present video is watched another time
- category block stats now inserts a new row each time that watched_db has a new addition
- category block stats decays the most recent row of every category in database each time rows are added.

Need to accept **CURRENT CHANGE** when resolving merge conflicts in dashboard.js and dashboard.html

Also, i think we need to make it decay the `current_level` when a video is NOT currently being watched, as when you spend some time away from youtube, and then come back to the home page, it will filter based on the values from the end of your last session - it would make sense that it decays over time while you are away. This however would involve autorunning the decay, which is outside of the scope of this issue and PR.

Can see below that videos with the same ID are inserted as a new row with a successive watch.

<img width="587" height="290" alt="image" src="https://github.com/user-attachments/assets/48d3a0cd-778d-4218-add5-23d69c0372af" />

If you follow the values in each of the categories, you can see how some of the values for a particular category decays while others have the value increased on watchtime.

<img width="767" height="608" alt="image" src="https://github.com/user-attachments/assets/05a65f61-3c1c-45bc-8a63-1db9725b2419" />

mouper committed:
Merge branch 'main' into tanish-khurana-patch-1

It-is-what-it-is1 committed:
refactor: adding missing functions

keanscott committed:
Revert "chore: category_block_stats now decays each category present in the database when watched_db is updated."

This reverts commit bef278b3edeebc127663a7cfb716978664dd7e36.

keanscott committed:
fix: removed allostatic functions from class -> does not work well with database actions.

keanscott committed:
fix: updated current_level increment and decay logic.

mouper committed:
fix: fix graph row updates, implement category decay scheduler, memory optimizations, graph improvements

Adds a background scheduler to periodically apply decay to all active categories based on their filter strength. Improves category block stats logic to use strength-based half-life, and exposes update functions for both all and specific categories. Optimizes YouTube extension memory usage with periodic cleanup, fixes watch logging to prevent duplicate sessions, and enhances graph rendering with gradients and timezone handling. Also updates settings route to trigger category decay on changes.

mouper committed:
Merge branch 'main' into tanish-khurana-patch-1

mouper committed:
feat: increase graph height
